ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/GSM.hs

```diff
@@ -81,34 +80,32 @@ semantics vars cmd = pre $ case cmd of
     ModifyCandidate peer bdel -> do
         atomically $ do
 
-            modifyTVar varIdlers $ Set.delete peer
-
-            v <- (Map.! peer) <$> readTVar varCandidates
-            Candidate b <- readTVar v
-            writeTVar v $! Candidate (b + bdel)
+            v <- (Map.! peer) <$> readTVar varStates
+            Candidate b <- psCandidate <$> readTVar v
+            writeTVar v $! PeerState (Candidate (b + bdel)) (Idling False)
 
         pure Unit
     NewCandidate peer bdel -> do
         atomically $ do
             Selection b _s <- readTVar varSelection
-            v <- newTVar $! Candidate (b + bdel)
-            modifyTVar varCandidates $ Map.insert peer v
+            v <- newTVar $! PeerState (Candidate (b + bdel)) (Idling False)
+            modifyTVar varStates $ Map.insert peer v
         pure Unit
     ReadGsmState -> do
         fmap ReadThisGsmState $ atomically $ readTVar varGsmState
     ReadMarker -> do
         fmap ReadThisMarker $ atomically $ readTVar varMarker
-    StartIdling peer -> do
-        atomically $ modifyTVar varIdlers $ Set.insert peer
+    StartIdling peer -> atomically $ do
```

Comment:
Would it make sense to guard against the case in which `peer` is not in the map?